Automattic/woocommerce-payments
# Add greeting to Account Balances header

Resolves #5721

### Changes proposed in this Pull Request









The work in this PR is currently not user-facing, it will only be rendered if the `_wcpay_feature_simplify_deposits_ui` feature flag is enabled.

This PR adds a greeting to the Account Balances card header, based on the time of day and the currently logged in user's first name.

The new Woo Payments logo has been added to the right side of the header.

![account-balances-header](https://user-images.githubusercontent.com/3147296/225180189-f8758d47-c934-44a8-9c76-96005d5ca5f2.png)

**Greeting time of day**

The greeting will adjust to the following based on the time of day. Source for these times: pdjTHR-2jn-p2#comment-3031.

Good morning [5am -11.59am]
Good afternoon [12pm – 4:59pm]
Good evening [5pm – 4:59am]

**Greeting name**
The name will match the logged-in user's "First Name" from `/wp-admin/profile.php`. If the "First Name" field is empty, the greeting will fall back to "Good {time of day} 👋".

### Designs

Design P2: paJDYF-7fi-p2

Design for this card header component:
![](https://user-images.githubusercontent.com/8490476/224234487-236b0aba-aa0e-4c08-a66c-88e192785fd8.png)


### Testing instructions





* Run `npm run build:client`.
* Enable the Simplify Deposits UI feature flag.  
  `update_option( '_wcpay_feature_simplify_deposits_ui', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
* Go to Payments → Overview.
* At the top of the Account Balances card, you should see the header with a greeting and Woo Payments logo that matches the design shown above.
* Greeting time of day: adjust your system's time (or wait) and ensure the correct time of day is displayed in the greeting, as per times in the description above.
* The greeting name should match the logged-in user's "First Name" from `/wp-admin/profile.php`.
* Set the logged-in user's "First Name" to an empty string via `/wp-admin/profile.php`. The greeting should fall back to "Good {time of day} 👋".
* The greeting should not visually break when provided a First Name with many characters, e.g. "Chrysanthemum".
* Ensure the header styles are responsive by reducing the viewport width or testing on a mobile device. Example of responsive styles on mobile with a long user "First Name":
   <img width="375" alt="image" src="https://user-images.githubusercontent.com/3147296/225454791-78641498-0cb7-4dac-aa9e-2ed95e95d3cf.png">
* Optional: Install the latest version of the Gutenberg plugin and ensure that the header styles look correct. (conflicting styles can occur with different versions of `@wordpress/components`)

-------------------


**Todo:**
- [x] Add greeting.
- [x] Add Woo Payments logo.
- [x] Lock-in vertical alignment across header greeting and logo
- [x] Make responsive.

**Post merge**
